src/core/bst.h

```diff
@@ -188,361 +133,540 @@ class BST {
     BST(BST_TYPE::BST_TYPE type) : type(type), lineno(0) {}
 #endif
     BST(BST_TYPE::BST_TYPE type, uint32_t lineno) : type(type), lineno(lineno) {}
-};
-
-class BST_expr : public BST {
-public:
-    virtual void* accept_expr(ExprVisitor* v) = 0;
 
-    BST_expr(BST_TYPE::BST_TYPE type) : BST(type) {}
-    BST_expr(BST_TYPE::BST_TYPE type, uint32_t lineno) : BST(type, lineno) {}
+    virtual bool isBST_dst() const { return false; }
 };
 
 class BST_stmt : public BST {
```

Comment:
It looks like this is the only class that inherits from BST now --  can we just fold these fields into BST itself?